### PR TITLE
Pin dnspython to avoid dependency incompatibility

### PR DESCRIPTION
**Pull request by Tawakalt**
**Proposed changes**:
- Fixes [ATO-1419](https://rasahq.atlassian.net/browse/ATO-1419)

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-1419]: https://rasahq.atlassian.net/browse/ATO-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -151,6 +151,9 @@ confluent-kafka = ">=1.9.2,<3.0.0"
 portalocker = "^2.7.0"
 structlog = "^23.1.0"
 structlog-sentry = "^2.0.2"
+# pin dnspython to avoid dependency incompatibility
```
Comment (ancalita) [inline pyproject.toml:154]: Can we mention the ticket ID for future reference in the comment

File: pyproject.toml [hunk 2]
```diff
@@ -151,6 +151,9 @@ confluent-kafka = ">=1.9.2,<3.0.0"
...
+dnspython = "2.3.0"
```
Comment (ancalita) [inline pyproject.toml:155]: Have you explored if changing the specification of `pymongo` in L299 to a version that locks dnspython to <2.4.2 works?

Comment (ancalita) [reply pyproject.toml:155]: If that pymongo version exists that is.

Comment (ancalita) [reply pyproject.toml:155]: Also do we need to backport to `3.6.x`?

Comment (Tawakalt) [reply pyproject.toml:155]: > If that pymongo version exists that is.

I didn't find any.

> Also do we need to backport to 3.6.x?

Yup, I'll do that once this is merged. I'll check 3.5.x and 3.4.x as well

**Pull request merged**